File: PAT/basic_level/1003_我要通过/pass_pat.c
#include <stdio.h>
#define MAX 101

int correct_cnt(const char *us_str, int *posi_p, int *posi_t);
int correct_order(const int posi_p, const int posi_t);
int correct_logic(const char *us_str);

int main(){
	int posi_p=0, posi_t=0;
	int i=0;
	int us_N=0;
	char us_str[MAX];

	scanf("%d", &us_N);
	if(us_N>=10){
		printf("NO\n");
		return 0;
	}

	for(i=0; i<us_N; i++){
		scanf("%s", us_str);
		if( correct_cnt(us_str, &posi_p, &posi_t) ){
			if( correct_order(posi_p, posi_t) ){
				if( correct_logic(us_str) ){
					printf("YES\n");
				}else{
					printf("NO\n");
				}
			}else{
				printf("NO\n");
			}
		}else{
			printf("NO\n");
		}
	}

	return 0;
}

int correct_cnt(const char *us_str, int *posi_p, int *posi_t){
	int cnt_p=0, cnt_a=0, cnt_t=0;
	int i=0;

	while( us_str[i]!='\0' ){
		if( us_str[i]=='P' ){
			cnt_p++;
			*posi_p = i;
		}else if( us_str[i]=='A' ){
			cnt_a++;
		}else if( us_str[i]=='T' ){
			cnt_t++;
			*posi_t = i;
		}else{
			return 0;
		}
		i++;
	}

	if( 1!=cnt_p ){
		return 0;
	}else if( 0==cnt_a ){
		return 0;
	}else if( 1!=cnt_t ){
		return 0;
	}else{
		return 1;
	}
}

int correct_order(const int posi_p, const int posi_t){
	return (posi_p+1) < posi_t;
}

int correct_logic(const char *us_str){
	int cnt_a1=0, cnt_a2=0, cnt_a3=0;
	int i=0;

	while( us_str[i++]!='P' ){
		cnt_a1++;
	}

	while( us_str[i++]!='T' ){
		cnt_a2++;
	}

	while( us_str[i++]!='\0' ){
		cnt_a3++;
	}

	if( cnt_a1*cnt_a2==cnt_a3 )
		return 1;
	else
		return 0;
}


/*
8
PAT
PAAT
AAPATAA
AAPAATAAAA
xPATx
PT
Whatever
APAAATAA


*/

File: PAT/basic_level/1003_我要通过/pass_pat.py
from __future__ import print_function

def correct_cnt(us_str):
	cnt_p, cnt_a, cnt_t = 0, 0, 0

	for i in range( len(us_str) ):
		if us_str[i]=='P':
			cnt_p += 1
			posi_p = i
		elif us_str[i]=='A':
			cnt_a += 1
		elif us_str[i]=='T':
			cnt_t += 1
			posi_t = i
		else:
			return False

	if 1!=cnt_p:
		return False
	elif 0==cnt_a:
		return False
	elif 1!=cnt_t:
		return False
	else:
		return True

def correct_order(us_str):
	return (us_str.find('P')+1) < us_str.find('T')

def correct_logic(us_str):
	a1, a2 = us_str.split('P')
	a2, a3 = a2.split('T')

	cnt_a = [ a1.count('A'), a2.count('A'), a3.count('A') ]

	if cnt_a[0]*cnt_a[1] == cnt_a[2]:
		return True
	else:
		return False

us_N = int(raw_input().strip())
if us_N<10:
	for i in range(us_N):
		us_str = raw_input()
		if correct_cnt(us_str):
			if correct_order(us_str):
				if correct_logic(us_str):
					print('YES')
				else:
					print('NO')
			else:
				print('NO')
		else:
			print('NO')
else:
	print('NO')


'''
8
PAT
PAAT
AAPATAA
AAPAATAAAA
xPATx
PT
Whatever
APAAATAA


'''






